contracts/lib/MessageBus.sol

```diff
@@ -869,8 +752,15 @@ library MessageBus {
         );
         bytes memory keyBytes = BytesLib.leftPad(bytes32ToBytes(_key));
         bytes memory path = BytesLib.concat(keyBytes, indexBytes);
-        bytes32 structPath = keccak256(abi.encodePacked(keccak256(
-                abi.encodePacked(path))));
+
+        bytes32 structPath = keccak256(
+            abi.encodePacked(
+                keccak256(
+                    abi.encodePacked(path)
+                )
+            )
+        );
```

Comment:
Is it correct to be hashed twice? 🤔

Reply:
Yes, It's correct.
 This was also tested with hackathon [mosaic.js ](https://github.com/OpenSTFoundation/mosaic.js/blob/94759dea024a4eee9ff4a0ce3658adc31656bd45/lib/setup/initStakeAndMint.js#L485)

JS version of this available [here](https://github.com/kedarchandrayan/mosaic.js/blob/94759dea024a4eee9ff4a0ce3658adc31656bd45/tools/proof/lib/helper.js#L29)

Reply:
Is it possibly a bug in the JS as well? What's the reason for hashing it twice?